sw/device/tests/sim_dv/i2c_tx_rx_test.c

```diff
@@ -0,0 +1,209 @@
+ */
+static volatile bool fmt_irq_seen = false;
+static volatile bool rx_irq_seen = false;
+static volatile bool done_irq_seen = false;
+
+void ottf_external_isr(void) {
+  // Find which interrupt fired at PLIC by claiming it.
+  dif_rv_plic_irq_id_t plic_irq_id;
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_claim(&plic, kTopEarlgreyPlicTargetIbex0, &plic_irq_id));
+
+  // Check if it is the right peripheral.
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+  CHECK(peripheral == kTopEarlgreyPlicPeripheralI2c0,
+        "Interurpt from unexpected peripheral: %d", peripheral);
+
+  // Correlate the interrupt fired.
+  dif_i2c_irq_t i2c_irq;
+  if (plic_irq_id == kTopEarlgreyPlicIrqIdI2c0FmtWatermark) {
+    fmt_irq_seen = true;
+    i2c_irq = kDifI2cIrqFmtWatermark;
+  } else if (plic_irq_id == kTopEarlgreyPlicIrqIdI2c0RxWatermark) {
+    rx_irq_seen = true;
+    i2c_irq = kDifI2cIrqRxWatermark;
+  } else if (plic_irq_id == kTopEarlgreyPlicIrqIdI2c0TransComplete) {
+    done_irq_seen = true;
+    i2c_irq = kDifI2cIrqTransComplete;
+  } else {
+    LOG_ERROR("Unexpected interrupt (at PLIC): %d", plic_irq_id);
+  }
+
+  // Check if the same interrupt fired at I2C as well.
+  bool is_pending;
+  CHECK_DIF_OK(dif_i2c_irq_is_pending(&i2c, i2c_irq, &is_pending));
+  CHECK(is_pending, "i2c interrupt fired at PLIC did not fire at I2C");
+
+  // Clear the interrupt at I2C.
+  CHECK_DIF_OK(dif_i2c_irq_acknowledge(&i2c, i2c_irq));
+
+  // Complete the IRQ at PLIC.
+  CHECK_DIF_OK(dif_rv_plic_irq_complete(&plic, kTopEarlgreyPlicTargetIbex0,
+                                        plic_irq_id));
+}
+
+bool test_main(void) {
+  CHECK_DIF_OK(
+      dif_i2c_init(mmio_region_from_addr(TOP_EARLGREY_I2C0_BASE_ADDR), &i2c));
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &plic));
+
+  CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(
+      &plic, kTopEarlgreyPlicIrqIdI2c0FmtWatermark, kTopEarlgreyPlicTargetIbex0,
+      kDifToggleEnabled));
+  CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(
+      &plic, kTopEarlgreyPlicIrqIdI2c0RxWatermark, kTopEarlgreyPlicTargetIbex0,
+      kDifToggleEnabled));
+  CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(
+      &plic, kTopEarlgreyPlicIrqIdI2c0TransComplete,
+      kTopEarlgreyPlicTargetIbex0, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_rv_plic_irq_set_priority(
+      &plic, kTopEarlgreyPlicIrqIdI2c0FmtWatermark, 0x1));
+  CHECK_DIF_OK(dif_rv_plic_irq_set_priority(
+      &plic, kTopEarlgreyPlicIrqIdI2c0RxWatermark, 0x1));
+  CHECK_DIF_OK(dif_rv_plic_irq_set_priority(
+      &plic, kTopEarlgreyPlicIrqIdI2c0TransComplete, 0x1));
+
+  // Enable the external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  // Temporary hack that connects i2c to a couple of open drain pins
```

Comment:
Don't forget

```c
pinmux_testutils_init(&pinmux);
```

Reply:
yeah so this area is going to conflict with a big PR Sri is doing. So all of this is a place holder until his big pinmux configuration drops.

Reply:
For what it's worth, the pinmux testutils init should not conflict with Sri's PR (or that PR is doing more than I remember us agreeing to). That function merely sets up the UART(s) and VBUS sense, and those pins aren't going to move. :)